### DISCUSSION TITLE
Debugging Meshes

### DISCUSSION THREAD

**Discussion by stewartoallen** [elalish/manifold#896] [General]
I'm using the JS/NPM package and would like to get more explicit error messages during Manifold construction. The error thrown from `new Instance.Manifold(mesh)` does not contain anything other than an integer code. Is it possible to get something like indexes to faces or edges that are failing the manifold test?

**Comment by pca006132** [comment]
Currently, no, and not even with the C++ API. We should indeed give more explicit error messages for this.

**Reply by elalish** [reply]
I *think* that integer is actually an enum to the error message, right? But yes, I've also thought about something more specific than "not manifold". Part of the issue is we early-out when we find a non-manifold edge; would it help to just have the first? Often if there's one, there are 10,000 others. If you look at our internal `IsManifold()` and `Is2Manifold()` functions, they are quite simple - if you want to debug the meshes you're producing, you may be better off recreating those for your own data structures.

**Reply by stewartoallen** [reply]
it may be useful to have a constructor option that produces a comprehensive list or a separate utility? I'm usually working with other people's externally created meshes. in future, I will end up writing my own validator.

**Reply by elalish** [reply]
PRs welcome!
